core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -181,11 +188,13 @@ class SocketServerTest extends JUnitSuite {
   def testSocketsCloseOnShutdown() {
     // open a connection
     val plainSocket = connect(protocol = SecurityProtocol.PLAINTEXT)
+    plainSocket.setTcpNoDelay(true)
     val traceSocket = connect(protocol = SecurityProtocol.TRACE)
+    traceSocket.setTcpNoDelay(true)
     val bytes = new Array[Byte](40)
     // send a request first to make sure the connection has been picked up by the socket server
-    sendRequest(plainSocket, bytes, Some(0))
-    sendRequest(traceSocket, bytes, Some(0))
+    sendRequest(plainSocket, bytes, Some(0), true)
+    sendRequest(traceSocket, bytes, Some(0), true)
```

Comment:
@ijuma see https://github.com/apache/kafka/pull/2632#issuecomment-283908626, I did this on purpose. I suffices to have a delay here when starting this up (tested over 10k+ iterations whereas disabling this fails after less than 50).

That said ... I added the wait inside the test method now and removed that complication. I think it's clearer if I just make it wait inside the test to not mask via slow_start and comment thing there.

Tested this solution over a ton of iterations too, works just as well and is much cleaner right?